### ISSUE TITLE
Foreground and background color transparency of widgets not rendering on opaque bar

### ISSUE THREAD

**Issue by KulkarniKaustubh** [qtile/qtile#4117]
### The issue:

```
$ qtile --version
0.22.1
```

The main issue here is that even though a widget's foreground or background color may have transparency in it (the Alpha part of it is not `ff`) the transparency is not visible when the bar's background is a non-transparent color.
Depending on the bar's background color's transparency, the widget's foreground or background color's transparency is being determined.

For example, I want my inactive groups to have a slightly transparent foreground color in my GroupBox widget. Hence, I have set up the widget in the following manner.

```
        widget.GroupBox(
            active="#add8e6",
            inactive="#add8e680",
            highlight_method="line",
            highlight_color=colors["dark_blue"],
            other_current_screen_border="#215578",
            other_screen_border="#215578",
            this_current_screen_border=colors["lighter_blue"],
            this_screen_border=colors["lighter_blue"],
            fontsize=15,
            font="Iosevka Nerd Font",
            background=backgrounds["group_box"],
            padding=10,
        ),
```

Though my `inactive` color has an `80` alpha value, it still renders as fully opaque. This is when the bar's background is fully opaque. Here is a screenshot:
![image](https://user-images.githubusercontent.com/55317079/212564157-5475a65a-757d-4b19-a0ea-11d06f66cf49.png)

Now, when my bar's background color is `#00000000`, the colors render perfectly. Here is the screenshot:
![image](https://user-images.githubusercontent.com/55317079/212564280-36e557b9-c2fb-4585-bf0f-1f3c65892321.png)

Is this expected behavior? I feel like regardless of the bar's background color, the widget's foreground/background color's alpha values should reflect in the bar.

Please do let me know how I could help fix this (if it does need fixing!

### Required:

- [X] I have searched past issues to see if this bug has already been reported.

**Comment by elParaguayo** [comment]
That is correct. Transparency is only enabled when the bar has a slightly transparent value. 

It doesn't have to be fully transparent (try an alpha value of FE for example)

**Comment by KulkarniKaustubh** [comment]
The problem with this is that my `widget.Systray` background looks really bad. Attaching screenshot:
![image](https://user-images.githubusercontent.com/55317079/212565048-92a9f775-46a9-44b2-9493-fcf30a2e6e51.png)

**Comment by elParaguayo** [comment]
Unfortunately, that is the sole reason why we disable transparency in bars with solid background. It's rubbish, I know.

Do your apps support the StatusNotifier spec? If so, you could try using that widget instead of `SysTray`.

**Comment by KulkarniKaustubh** [comment]
Understood. Will try looking into the StatusNotifier widget!

Thank you for your time!

**Comment by elParaguayo** [comment]
#4126 will address this. Here's an example:
![20230121_091937](https://user-images.githubusercontent.com/946265/214955227-7de0d5e4-0a85-4220-a8db-628aabca31a1.png)

**Comment by ramnes** [comment]
Is it the same as #4510?

**Comment by elParaguayo** [comment]
Not really. That one is really asking about how a bar with a gradient impacts the widgets that are rendered in that bar.

**Comment by skela** [comment]
Also have the same issue, StatusNotifier doesnt really seem to work for most of the apps i have.

Systray
![2024-07-03_23-55](https://github.com/qtile/qtile/assets/58268/ce17e51c-d6f2-43a3-b65b-1ba9a6d9ce54)

StatusNotifier
![2024-07-03_23-56](https://github.com/qtile/qtile/assets/58268/fd4cd8c6-dd0a-4403-afff-ae47ccdb7da9)

**Comment by elParaguayo** [comment]
@skela I don't think that's the same issue at all. Your issue seems to be that some apps aren't showing an icon, not that the background isn't trending correctly.

Two things:
1) StatusNotifier isn't a drop-in replacement for Systray. They use different specs so some apps will show icons with one and not the other. Some apps will work with both.
2) there could well be bugs with the StatusNotifier widget so please check log for errors and let me know if there are any apps that you know support the spec but aren't working.

**Comment by skela** [comment]
Its not exactly the same, but is it not similar enough to be considered as part of this issue? I agree with @KulkarniKaustubh 's comment:
"Is this expected behavior? I feel like regardless of the bar's background color, the widget's foreground/background color's alpha values should reflect in the bar."

Perhaps with the modification that if I give my Widget a background color, I expect my widget to have that color.
![2024-07-04_08-18](https://github.com/qtile/qtile/assets/58268/c7c0021b-5883-4e1b-a367-aa2f3e3c5068)

I'll give the StatusNotifier another go and try to find some logs.

**Comment by elParaguayo** [comment]
For Systray this is not currently possible. This behaviour is expected and explained in our docs.

That's an issue with how system trays work in x11. To fix it,we need to inplement psuedotransparency which means, rather than having a transparent window for the bar, we use an opaque one but copy the wallpaper to it. This is hard. We've started trying to implement it but are still finding bugs with it.
